### ISSUE TITLE
Directory resource fails due to Insufficient Permissions

### ISSUE THREAD

**Issue by smurawski** [chef/chef#3164]
ChefDK 0.4.0 with Chef Client 12.0.3

Directory resource fails to create directories in some folder paths - even though permissions do not appear different.

```powershell
#Worked 
chef-apply --% -e 'directory "c:/users/administrator/test"'
#Failed
chef-apply --% -e 'directory "c:/users/administrator/desktop/test'
#Worked
chef-apply --% -e 'directory "c:/test'
#Failed
chef-apply --% -e 'directory "c:/users/test'
```

Originally reported against ChefDK - https://github.com/chef/chef-dk/issues/340

**Comment by jdmundrawala** [comment]
Related to https://github.com/chef/chef/pull/2896 ?

**Comment by smurawski** [comment]
Possibly.. Tracking it down now

**Comment by smurawski** [comment]
I tested with 
```powershell
 chef-apply --% -e 'directory "c:/Users/Administrator/Desktop/test"'
```
against Server 2012 with 
Chef Client 11.6.0 - FAILED
Chef Client 11.18.6 - FAILED
Chef Client 12.0.3 - FAILED
Chef Client 12.1.2 - SUCCESS

I'll do some more complete testing, but this only affects certain directories (principally, maybe only, in the local Administrator user profile).

**Comment by smurawski** [comment]
@jdmundrawala Yeah, looks very much related to that PR, since it is resolved in 12.1.x
